On Solaris /usr/include/sys/isa_defs.h defines _LITTLE_ENDIAN or _BIG_ENDIAN
without any value.

https://github.com/aappleby/smhasher/pull/55

diff -r 08f2f4958b49 -r 9d8c77cdb4d0 gfx/angle/checkout/src/common/third_party/smhasher/src/PMurHash.cpp
--- a/gfx/angle/checkout/src/common/third_party/smhasher/src/PMurHash.cpp	Wed Apr 24 14:38:42 2019 +0200
+++ b/gfx/angle/checkout/src/common/third_party/smhasher/src/PMurHash.cpp	Wed Apr 24 17:15:22 2019 +0200
@@ -92,9 +92,9 @@
  * or even _LITTLE_ENDIAN or _BIG_ENDIAN (Note the single _ prefix) */
 #if !defined(__BYTE_ORDER)
 #    if defined(__LITTLE_ENDIAN__) && __LITTLE_ENDIAN__ == 1 || \
-        defined(_LITTLE_ENDIAN) && _LITTLE_ENDIAN == 1
+        defined(_LITTLE_ENDIAN)
 #        define __BYTE_ORDER __LITTLE_ENDIAN
-#    elif defined(__BIG_ENDIAN__) && __BIG_ENDIAN__ == 1 || defined(_BIG_ENDIAN) && _BIG_ENDIAN == 1
+#    elif defined(__BIG_ENDIAN__) && __BIG_ENDIAN__ == 1 || defined(_BIG_ENDIAN)
 #        define __BYTE_ORDER __BIG_ENDIAN
 #    endif
 #endif
